core/eolearn/core/eodata_io.py

```diff
@@ -766,25 +774,32 @@ def save(
         filesystem: FS,
         feature_path: str,
         compress_level: int = 0,  # noqa: ARG003
-        temporal_selection: None | slice | list[int] = None,
+        temporal_selection: None | InferredSelection = None,
     ) -> str:
         cls._check_dependencies_imported(feature_path)
         path = feature_path + cls.get_file_extension()
         store = cls._get_mapping(path, filesystem)
+        chunk_size = (1, *data.shape[1:])
 
-        if temporal_selection is None:
-            chunk_size = (1, *data.shape[1:])
+        if temporal_selection is None or temporal_selection.selection is None:
             zarr.save_array(store, data, chunks=chunk_size)
             return path
 
+        selection, full_size = temporal_selection.selection, temporal_selection.full_size
+
         try:
             zarray = zarr.open_array(store, "r+")
         except ValueError as error:  # zarr does not expose the proper error...
-            raise IOError(
-                f"Unable to open Zarr array at {path!r}. Saving with `temporal_selection` requires preexisting zarr."
-            ) from error
+            if full_size is not None:
+                zarr_shape = (full_size, *data.shape[1:])
```

Comment:
`zarr_shape` and `chunk_size` are very similar; perhaps you could have a var `data_shape = *data.shape[1:]` and then use `(full_size, data_shape)` and `(1, data_shape)` respectively. 

Just an idea, no need to change.

Reply:
i was considering it, but i'd keep it as it is for now since it's not that much simpler